Extensions/Lighting/JsExtension.js

```diff
@@ -290,74 +290,31 @@ module.exports = {
           .getValue()
       );
       if (this._radius <= 0) this._radius = 1;
-      const colorHex = objectsRenderingService.rgbOrHexToHexNumber(
+      const color = objectsRenderingService.rgbOrHexToHexNumber(
         this._associatedObjectConfiguration
           .getProperties(this.project)
           .get('color')
           .getValue()
       );
-      this._color = [
-        ((colorHex >> 16) & 0xff) / 255,
-        ((colorHex >> 8) & 0xff) / 255,
-        (colorHex & 0xff) / 255,
-      ];
-
-      const geometry = new PIXI.Geometry();
-      const shader = PIXI.Shader.from(
-        `
-    precision mediump float;
-    attribute vec2 aVertexPosition;
-
-    uniform mat3 translationMatrix;
-    uniform mat3 projectionMatrix;
-    varying vec2 vPos;
-
-    void main() {
-        vPos = aVertexPosition;
-        gl_Position = vec4((projectionMatrix * translationMatrix * vec3(aVertexPosition, 1.0)).xy, 0.0, 1.0);
-    }`,
-        `
-    precision mediump float;
-    uniform vec2 center;
-    uniform float radius;
-    uniform vec3 color;
-    uniform mat3 translationMatrix;
-    uniform mat3 projectionMatrix;
-
-    varying vec2 vPos;
-
-    void main() {
-        float l = length(vPos - center);
-        float intensity = 0.0;
-        if(l < radius)
-          intensity = clamp((radius - l)*(radius - l)/(radius*radius), 0.0, 1.0);
-        gl_FragColor = vec4(color*intensity, 1.0);
-    }
-    `,
-        {
-          center: [this._instance.getX(), this._instance.getY()],
-          radius: this._radius,
-          color: this._color,
-        }
+
+      // The icon in the middle.
+      const lightIconSprite = new PIXI.Sprite(PIXI.Texture.from('CppPlatform/Extensions/lightIcon32.png'));
```

Comment:
Can you use `pixiResourcesLoader` to load the texture?

Reply:
I wondered but I'm not sure if we should because this is not a resource of the game, bit a built-in image from the editor itself.